# FUNDING ANALYSIS FOR INDIAN STARTUPS

#### Team: Team Namibia

## Table of Contents


- [Business Understanding](#Business-Understanding)

- [Data Understanding](#data-understanding)
    - [Data Collection](#data-collection)
    - [Feature Description](#feature-description)
    - [Viewing the data](#viewing-the-data)
    - [Data Quality Assessment](#data-quality-assessment)

- [Data Preparation](#data-preparation)
    - [Data Cleaning](#data-cleaning)
    - [Exploratory Data Analysis](#exploratory-data-analysis)



## Business Understanding
Team Namibia aims to venture into the start-up space in India and being the data expert of the team, we've have been tasked with investigating the econmoic landscape and coming out with a course of action for this endeavour.


#### Problem Statement:
Investigating the dynamics of startup funding in India over the period from 2018 to 2021. The aim is to understand the trends, sector preferences, investment stages, key investors, and geographical distribution of the funding. Additionally, if there have been significant differences in funding amounts across different years and sectors, it can guide the action plan to be taken.

#### Objective
The goal of this analysis is to provide insights into the startup funding landscape in India from 2018 to 2021. 
- Identify trends and patterns in funding amounts over the years.
- Determine which sectors received the most funding and how sector preferences changed over time.
- Understand the distribution of funding across different stages of startups (e.g., Seed, Series A).
- Identify key investors and their investment behaviors.
- Analyze the geographical distribution of funding within India.

#### Analytical Questions
1. What are the trends and patterns in funding amounts for startups in India beetween 2018 to 2021?
   - Analyzing the annual and quarterly trends in funding can reveal patterns and growth trajectories. Look for peaks, dips, and any consistent growth patterns over these years.
2. Which sectors received the most funding, and how did sector preferences change over time from 2018 to 2021?
   - Identifying which industries or sectors received the most funding can show sectoral preferences and shifts. Understanding how this distribution has evolved over the years can highlight emerging trends and declining interests.
3. How is the distribution of funding across different stages of startups (e.g., Seed, Series A)?
   - Analyzing the funding amounts at different startup stages can provide insights into the investment appetite at various growth phases. It can also help in understanding the maturity and risk preference of investors.
4. Who are the key investors in Indian startups, and what are their investment behaviors/patterns?
   - Identifying the most active investors and analyzing their investment portfolios can shed light on key players in the ecosystem. Understanding their investment patterns can also reveal strategic preferences and alliances.
5. What is the geographical distribution of startup funding within India, and how has this distribution changed over the years 2018 to 2021?
   - Analyzing the geographical distribution of startup funding can show regional hotspots for entrepreneurship and investment. Observing how this has changed over the years can reveal shifts in regional focus and development.



### Null Hypothesis (H0)
There is no significant difference in the funding amounts for startups in India between the years 2018 and 2021 across different sectors.

### Alternate Hypothesis (H1)
There is a significant difference in the funding amounts for startups in India between the years 2018 and 2021 across different sectors.


### Importing the necessary packages 

In [83]:
# Import the pyodbc library to handle ODBC database connections
import pyodbc 

# Import the dotenv function to load environment variables from a .env file
from dotenv import dotenv_values 

# Import the pandas and numpy library for data manipulation and analysis
import pandas as pd 
import numpy as np


# Importing Matplotlib for  visualizations in Python
import matplotlib.pyplot as plt

# Importing Seaborn for statistical data visualization based on Matplotlib
import seaborn as sns


# Import the warnings library to handle warning messages
import warnings

# Filter out (ignore) any warnings that are raised
warnings.filterwarnings('ignore')


 ### Establishing a connection to the database

In [14]:
# Load environment variables from .env file into a dictionary
environment_variables = dotenv_values('.env')

# Get the values for the credentials you set in the .env file
database = environment_variables.get("DATABASE")
server = environment_variables.get("SERVER")
username = environment_variables.get("UID")
password = environment_variables.get("PWD")

# Create the connection string using the retrieved credentials
connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password};MARS_Connection=yes;MinProtocolVersion=TLSv1.2;"



In [15]:
# Establish a connection to the database using the connection string
connection = pyodbc.connect(connection_string) 

In [16]:
# Define the SQL query to select all columns from the specified table
query = "Select * from dbo.LP1_startup_funding2020"

# Execute the SQL query and fetch the result into a pandas DataFrame using the established database connection
data1 = pd.read_sql(query, connection)

data1

Company_Brand  Founded HeadQuarter              Sector  \
0       Aqgromalin   2019.0     Chennai            AgriTech   
1         Krayonnz   2019.0   Bangalore              EdTech   
2     PadCare Labs   2018.0        Pune  Hygiene management   
3            NCOME   2020.0   New Delhi              Escrow   
4       Gramophone   2016.0      Indore            AgriTech   
...            ...      ...         ...                 ...   
1050  Leverage Edu      NaN       Delhi              Edtech   
1051         EpiFi      NaN        None             Fintech   
1052       Purplle   2012.0      Mumbai           Cosmetics   
1053        Shuttl   2015.0       Delhi           Transport   
1054         Pando   2017.0     Chennai            Logitech   

                                           What_it_does  \
0                          Cultivating Ideas for Profit   
1     An academy-guardian-scholar centric ecosystem ...   
2      Converting bio-hazardous waste to harmless waste   
3                          Escrow-as-a-service platform   
4     Gramophone is an AgTech platform enabling acce...   
...                                                 ...   
1050  AI enabled marketplace that provides career gu...   
1051  It offers customers with a single interface fo...   
1052         Online makeup and beauty products retailer   
1053                    App based bus aggregator serice   
1054            Networked logistics management software   

                                               Founders  \
0                       Prasanna Manogaran, Bharani C L   
1                      Saurabh Dixit, Gurudutt Upadhyay   
2                                       Ajinkya Dhariya   
3                                         Ritesh Tiwari   
4     Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
...                                                 ...   
1050                                  Akshay Chaturvedi   
1051                    Sujith Narayanan, Sumit Gwalani   
1052                          Manish Taneja, Rahul Dash   
1053                      Amit Singh, Deepanshu Malviya   
1054                     Jayakrishnan, Abhijeet Manohar   

                                   Investor      Amount       Stage column10  
0                           Angel investors    200000.0        None     None  
1                           GSF Accelerator    100000.0    Pre-seed     None  
2                            Venture Center         NaN    Pre-seed     None  
3       Venture Catalysts, PointOne Capital    400000.0        None     None  
4       Siana Capital Management, Info Edge    340000.0        None     None  
...                                     ...         ...         ...      ...  
1050  DSG Consumer Partners, Blume Ventures   1500000.0        None     None  
1051          Sequoia India, Ribbit Capital  13200000.0  Seed Round     None  
1052                             Verlinvest   8000000.0        None     None  
1053             SIG Global India Fund LLP.   8043000.0    Series C     None  
1054                      Chiratae Ventures   9000000.0    Series A     None  

[1055 rows x 10 columns]

In [17]:
# Define the SQL query to select all columns from the specified table
query = "Select * from dbo.LP1_startup_funding2021"

# Execute the SQL query and fetch the result into a pandas DataFrame using the established database connection
data2 = pd.read_sql(query, connection)

data2

Company_Brand  Founded HeadQuarter                 Sector  \
0     Unbox Robotics   2019.0   Bangalore             AI startup   
1             upGrad   2015.0      Mumbai                 EdTech   
2        Lead School   2012.0      Mumbai                 EdTech   
3            Bizongo   2015.0      Mumbai         B2B E-commerce   
4           FypMoney   2021.0    Gurugram                FinTech   
...              ...      ...         ...                    ...   
1204        Gigforce   2019.0    Gurugram  Staffing & Recruiting   
1205          Vahdam   2015.0   New Delhi       Food & Beverages   
1206    Leap Finance   2019.0   Bangalore     Financial Services   
1207    CollegeDekho   2015.0    Gurugram                 EdTech   
1208          WeRize   2019.0   Bangalore     Financial Services   

                                           What_it_does  \
0     Unbox Robotics builds on-demand AI-driven ware...   
1        UpGrad is an online higher education platform.   
2     LEAD School offers technology based school tra...   
3     Bizongo is a business-to-business online marke...   
4     FypMoney is Digital NEO Bank for Teenagers, em...   
...                                                 ...   
1204                  A gig/on-demand staffing company.   
1205  VAHDAM is among the world’s first vertically i...   
1206  International education loans for high potenti...   
1207  Collegedekho.com is Student’s Partner, Friend ...   
1208  India’s first socially distributed full stack ...   

                                               Founders  \
0                           Pramod Ghadge, Shahid Memon   
1     Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                            Smita Deorah, Sumeet Mehta   
3               Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                         Kapil Banwari   
...                                                 ...   
1204                        Chirag Mittal, Anirudh Syal   
1205                                         Bala Sarda   
1206                         Arnav Kumar, Vaibhav Singh   
1207                                       Ruchir Arora   
1208                      Vishal Chopra, Himanshu Gupta   

                                           Investor        Amount  \
0                       BEENEXT, Entrepreneur First    $1,200,000   
1          Unilazer Ventures, IIFL Asset Management  $120,000,000   
2                  GSV Ventures, Westbridge Capital   $30,000,000   
3                            CDC Group, IDG Capital   $51,000,000   
4     Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000   
...                                             ...           ...   
1204                                Endiya Partners      $3000000   
1205                                       IIFL AMC     $20000000   
1206                                   Owl Ventures     $55000000   
1207               Winter Capital, ETS, Man Capital     $26000000   
1208                 3one4 Capital, Kalaari Capital      $8000000   

             Stage  
0     Pre-series A  
1             None  
2         Series D  
3         Series C  
4             Seed  
...            ...  
1204  Pre-series A  
1205      Series D  
1206      Series C  
1207      Series B  
1208      Series A  

[1209 rows x 9 columns]

In [20]:
# Write the concatenated DataFrame 'df' to a CSV file named 'lp1.csv'
data1.to_csv('startup_funding2020.csv')
data2.to_csv('startup_funding2021.csv')

In [18]:
# Concatenate two DataFrames 'data1' and 'data2' vertically (along the rows)
df = pd.concat([data1, data2])

# Write the concatenated DataFrame 'df' to a CSV file named 'lp1.csv'
df.to_csv('lp1.csv')

# Read the CSV file into a DataFrame
lp1 = pd.read_csv('lp1.csv')

lp1.head(10)


Unnamed: 0 Company_Brand  Founded HeadQuarter               Sector  \
0           0    Aqgromalin   2019.0     Chennai             AgriTech   
1           1      Krayonnz   2019.0   Bangalore               EdTech   
2           2  PadCare Labs   2018.0        Pune   Hygiene management   
3           3         NCOME   2020.0   New Delhi               Escrow   
4           4    Gramophone   2016.0      Indore             AgriTech   
5           5        qZense   2019.0   Bangalore             AgriTech   
6           6  MyClassboard   2008.0   Hyderabad               EdTech   
7           7         Metvy   2018.0     Gurgaon  Networking platform   
8           8        Rupeek   2015.0   Bangalore              FinTech   
9           9     Gig India   2017.0        Pune        Crowdsourcing   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   
5  qZense Labs is building the next-generation Io...   
6  MyClassboard is a full-fledged School / Colleg...   
7  AI driven networking platform for individuals ...   
8  Rupeek is an online lending platform that spec...   
9  GigIndia is a marketplace that provides on-dem...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
5                       Rubal Chib, Dr Srishti Batra   
6                                     Ajay Sakhamuri   
7                                   Shawrya Mehrotra   
8            Amar Prabhu, Ashwin Soni, Sumit Maniyar   
9                       Aditya Shirole, Sahil Sharma   

                                        Investor      Amount         Stage  \
0                                Angel investors    200000.0           NaN   
1                                GSF Accelerator    100000.0      Pre-seed   
2                                 Venture Center         NaN      Pre-seed   
3            Venture Catalysts, PointOne Capital    400000.0           NaN   
4            Siana Capital Management, Info Edge    340000.0           NaN   
5  Venture Catalysts, 9Unicorns Accelerator Fund    600000.0          Seed   
6                                    ICICI Bank.    600000.0  Pre-series A   
7                                     HostelFund         NaN    Pre-series   
8   KB Investment, Bertelsmann India Investments  45000000.0      Series C   
9      Shantanu Deshpande, Subramaniam Ramadorai   1000000.0  Pre-series A   

  column10  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN  
5      NaN  
6      NaN  
7      NaN  
8      NaN  
9      NaN

In [19]:
data1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   Founded        842 non-null    float64
 2   HeadQuarter    961 non-null    object 
 3   Sector         1042 non-null   object 
 4   What_it_does   1055 non-null   object 
 5   Founders       1043 non-null   object 
 6   Investor       1017 non-null   object 
 7   Amount         801 non-null    float64
 8   Stage          591 non-null    object 
 9   column10       2 non-null      object 
dtypes: float64(2), object(8)
memory usage: 82.5+ KB


[Back to Top](#Table-of-Contents)


# Data Understanding

In this section, we seek to undestand the data that will help us solve our business problem/question. We detail how the briefly mention how the data was collected, we describe the useful features within the dataset, assess our data through an exploratory data analysis and then conclude with a data quality assessment

### Data Collection
The analysis for the Indian startup funding covers a span of 4 years and the data on each of the years comes via a unique dataset in a csv format. 
- Two of the datasets (2020 and 2021) were obtained from the sakila dataset online. The 2020 dataset was saved to a variable data1 whilst the 2021 dataset was saved to data2.
- The dataset for the year 2019 is a  file named "startup_funding2019.csv" obtained from OneDrive using the link https://azubiafrica-my.sharepoint.com/personal/teachops_azubiafrica_org/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fteachops%5Fazubiafrica%5Forg%2FDocuments%2FCareer%20Accelerator%20Data%5FSets%2FLP1%20Datasets&ga=1 
- The final dataset is a file with the name "startup_funding2018.csv" and was obtained from a Github Repository that can be accessed uusing this link https://github.com/Azubi-Africa/Career_Accelerator_LP1-Data_Analysis.


### Feature Description
Descrpition of the columns in the dataset: 

- Company/Brand: Name of the company/start-up
- Founded: Year start-up was established
- Headquarters/Location: Location where the start-up is headquartered
- Sector/Industry: The industry or sector in which the start-up operates, such as healthtech, fintech, etc.
- What it does/About Company: Brief overview of the company's function
- Founders: Founders of the Company
- Amount: Total amount raised by the start-up in each funding round
- Stage/Round: Details of the funding stages such as seed, series A, series B, etc.
- Investors: The names of the investors or investment firms involved

These are the useful variables that will provide the nexxesary data to answer our analytical business questions.




### Viewing the data
Here we will take a look into the data using different methods to gain an overall perspective of the data structure (number of columns and rows), the  number of columns and their data types and a sample of the dataframe (a preview of a few rows and columns)

In [57]:
# Read data from the CSV file 'startup_funding2020.csv' into a DataFrame named d21
d20 = pd.read_csv('startup_funding2020.csv')

# Display the first few rows of the DataFrame d20
d20.head()



Company_Brand  Founded HeadQuarter              Sector  \
0    Aqgromalin   2019.0     Chennai            AgriTech   
1      Krayonnz   2019.0   Bangalore              EdTech   
2  PadCare Labs   2018.0        Pune  Hygiene management   
3         NCOME   2020.0   New Delhi              Escrow   
4    Gramophone   2016.0      Indore            AgriTech   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   

                              Investor    Amount     Stage  
0                      Angel investors  200000.0       NaN  
1                      GSF Accelerator  100000.0  Pre-seed  
2                       Venture Center       NaN  Pre-seed  
3  Venture Catalysts, PointOne Capital  400000.0       NaN  
4  Siana Capital Management, Info Edge  340000.0       NaN

In [27]:
# Print information about the structure of the DataFrame d20
d20.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0     1055 non-null   int64  
 1   Company_Brand  1055 non-null   object 
 2   Founded        842 non-null    float64
 3   HeadQuarter    961 non-null    object 
 4   Sector         1042 non-null   object 
 5   What_it_does   1055 non-null   object 
 6   Founders       1043 non-null   object 
 7   Investor       1017 non-null   object 
 8   Amount         801 non-null    float64
 9   Stage          591 non-null    object 
 10  column10       2 non-null      object 
dtypes: float64(2), int64(1), object(8)
memory usage: 90.8+ KB


In [29]:
# Display the shape of the DataFrame d20 (number of rows, number of columns)
d20.shape

(1055, 11)

In [37]:
# Generate summary statistics for the numerical columns in DataFrame d20 
num_stats = d20.describe().T

num_stats


count          mean           std      min        25%        50%  \
Unnamed: 0  1055.0  5.270000e+02  3.046966e+02      0.0      263.5      527.0   
Founded      842.0  2.015363e+03  4.097909e+00   1973.0     2014.0     2016.0   
Amount       801.0  1.130430e+08  2.476635e+09  12700.0  1000000.0  3000000.0   

                   75%           max  
Unnamed: 0       790.5  1.054000e+03  
Founded         2018.0  2.020000e+03  
Amount      11000000.0  7.000000e+10

In [39]:
# Generate descriptive statistics for the categorical columns in DataFrame d20
categorical_stats = d20.describe(include=['object']).T
categorical_stats

count unique                               top freq
Company_Brand  1055    905                             Nykaa    6
HeadQuarter     961     77                         Bangalore  317
Sector         1042    302                           Fintech   80
What_it_does   1055    990  Provides online learning classes    4
Founders       1043    927                     Falguni Nayar    6
Investor       1017    848                 Venture Catalysts   20
Stage           591     42                          Series A   96
column10          2      2                          Pre-Seed    1

In [40]:
# Check for the total number of missing values in each column of the DataFrame d20
d20.isna().sum()


Unnamed: 0          0
Company_Brand       0
Founded           213
HeadQuarter        94
Sector             13
What_it_does        0
Founders           12
Investor           38
Amount            254
Stage             464
column10         1053
dtype: int64

In [42]:
# Check for duplicates in the d20 Dataframe
d20.duplicated().sum()



0

#### Data Quality Assessment
With respect to Data Quality Assessment, it is where the quality of the data is assessed, including completeness, accuracy, consistency and relevance. This is a crucial step in understanding the shortcomings of our data and being able to use that knwoledge to plan how to handle/clean the data.

The dataset "startup_funding2020" has 11 columns and 1055 rows but has no duplicate rows. The columns in the dataset are a mixture of strings/text (Company Brand, Sector, Headquarters, Founders, What it does, Investors, Stage, Column10), floats (Amount, Founded) and integers (Unnamed : 0). 

In this dataset, 3 columns had no missing values (Unnamed :0, Company Brand and What it does) with majority of the columns having less than 10% missing values (HeadQuarter, Sector, Founders, Investor). Only two other columns had over 25% missing values which were stage (43%) and Column10 (99.81%). 

It is instantly obvious the columns Unnamed :0 and column 10 have no relevance cause they possess no useful data with respect to our project goals. Other columns that would not be needed in reaching our goals (answering the analytical business questions) are Founded and What_It_Does.

[Back to Top](#Table-of-Contents)

### Data Cleaning
Messy data such as missing data and data in the wrong format can greatly impair our ability to analyze our data and gain useful insights from it. So at this stage, we clean the data so it can be useful in our analysis. 

#### Handling Missing Values
The unecessary columns are dropped from the dataset. For the two remaining columns with over 25% missing values (Stage and Amount), the rows that have missing values in both columns are also removed. The resulting dataframe is assessed for the new levels of missing data. For the columns wit less than 10% missing values, they are willed with either the median or mode.

In [46]:
#Viewing the missing values in each column of the dataframe d20
d20.isna().sum()

Unnamed: 0          0
Company_Brand       0
Founded           213
HeadQuarter        94
Sector             13
What_it_does        0
Founders           12
Investor           38
Amount            254
Stage             464
column10         1053
dtype: int64

In [65]:
#dropping the unecessary columns
d20.drop(columns=['column10', 'Unnamed: 0', 'Founded', 'What_it_does'], inplace= True)
d20

Company_Brand HeadQuarter              Sector  \
0       Aqgromalin     Chennai            AgriTech   
1         Krayonnz   Bangalore              EdTech   
2     PadCare Labs        Pune  Hygiene management   
3            NCOME   New Delhi              Escrow   
4       Gramophone      Indore            AgriTech   
...            ...         ...                 ...   
1050  Leverage Edu       Delhi              Edtech   
1051         EpiFi         NaN             Fintech   
1052       Purplle      Mumbai           Cosmetics   
1053        Shuttl       Delhi           Transport   
1054         Pando     Chennai            Logitech   

                                               Founders  \
0                       Prasanna Manogaran, Bharani C L   
1                      Saurabh Dixit, Gurudutt Upadhyay   
2                                       Ajinkya Dhariya   
3                                         Ritesh Tiwari   
4     Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
...                                                 ...   
1050                                  Akshay Chaturvedi   
1051                    Sujith Narayanan, Sumit Gwalani   
1052                          Manish Taneja, Rahul Dash   
1053                      Amit Singh, Deepanshu Malviya   
1054                     Jayakrishnan, Abhijeet Manohar   

                                   Investor      Amount       Stage  
0                           Angel investors    200000.0         NaN  
1                           GSF Accelerator    100000.0    Pre-seed  
2                            Venture Center         NaN    Pre-seed  
3       Venture Catalysts, PointOne Capital    400000.0         NaN  
4       Siana Capital Management, Info Edge    340000.0         NaN  
...                                     ...         ...         ...  
1050  DSG Consumer Partners, Blume Ventures   1500000.0         NaN  
1051          Sequoia India, Ribbit Capital  13200000.0  Seed Round  
1052                             Verlinvest   8000000.0         NaN  
1053             SIG Global India Fund LLP.   8043000.0    Series C  
1054                      Chiratae Ventures   9000000.0    Series A  

[1055 rows x 7 columns]

In [66]:

# Filter rows where either the 'Amount' and 'Name' column has missing values
d31= d20[ d20['Amount'].isna() & d20['Stage'].isna()]
#d31.shape

# Checking the missing values for the new Dataframe d30
d31


Company_Brand HeadQuarter              Sector  \
12              MasterG   New Delhi     Fashion startup   
18            Pine Labs       Noida             FinTech   
29            Delhivery     Gurgaon          E-commerce   
40         Fleeca India      Jaipur     Tyre management   
44     PointOne Capital   Bangalore  Venture capitalist   
...                 ...         ...                 ...   
1021        IncubateHub   Bengaluru            Tech hub   
1022        Rage Coffee       Delhi                FMCG   
1023          Skilancer       Noida          Technology   
1024  Harappa Education   New Delhi              Edtech   
1028            PayMart  Chandigarh             Fintech   

                                               Founders  \
12                                        Gayatri Jolly   
18                                           Amrish Rau   
29    Kapil Bharati, Mohit Tandon, Sahil Barua, Sura...   
40                                           Tikam Jain   
44                                            Mihir Jha   
...                                                 ...   
1021                                    Rajiv Mukherjee   
1022                                       Bharat Sethi   
1023                                       Neeraj Kumar   
1024                                  Pramath Raj Sinha   
1028                                        Amit Narang   

                                               Investor  Amount Stage  
12                        Acumen Fund's angel programme     NaN   NaN  
18                         Mastercard, Temasek Holdings     NaN   NaN  
29    Steadview Capital, Canada Pension Plan Investm...     NaN   NaN  
40                                    Bridgestone India     NaN   NaN  
44                                                  NaN     NaN   NaN  
...                                                 ...     ...   ...  
1021                                  Venture Catalysts     NaN   NaN  
1022                                      Refex Capital     NaN   NaN  
1023                                  Venture Catalysts     NaN   NaN  
1024                     James Murdoch-led Lupa Systems     NaN   NaN  
1028  IIM Ahmedabad’s incubator (CIIE.CO),Chandigarh...     NaN   NaN  

[141 rows x 7 columns]

In [67]:
#dropping the missing rows for both Amount and Stage from the dataframe d20 and saving it as dataframe d20_Cleaned1
d20_cleaned1= d20.drop(index=d31.index)
d20_cleaned1

Company_Brand HeadQuarter              Sector  \
0       Aqgromalin     Chennai            AgriTech   
1         Krayonnz   Bangalore              EdTech   
2     PadCare Labs        Pune  Hygiene management   
3            NCOME   New Delhi              Escrow   
4       Gramophone      Indore            AgriTech   
...            ...         ...                 ...   
1050  Leverage Edu       Delhi              Edtech   
1051         EpiFi         NaN             Fintech   
1052       Purplle      Mumbai           Cosmetics   
1053        Shuttl       Delhi           Transport   
1054         Pando     Chennai            Logitech   

                                               Founders  \
0                       Prasanna Manogaran, Bharani C L   
1                      Saurabh Dixit, Gurudutt Upadhyay   
2                                       Ajinkya Dhariya   
3                                         Ritesh Tiwari   
4     Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
...                                                 ...   
1050                                  Akshay Chaturvedi   
1051                    Sujith Narayanan, Sumit Gwalani   
1052                          Manish Taneja, Rahul Dash   
1053                      Amit Singh, Deepanshu Malviya   
1054                     Jayakrishnan, Abhijeet Manohar   

                                   Investor      Amount       Stage  
0                           Angel investors    200000.0         NaN  
1                           GSF Accelerator    100000.0    Pre-seed  
2                            Venture Center         NaN    Pre-seed  
3       Venture Catalysts, PointOne Capital    400000.0         NaN  
4       Siana Capital Management, Info Edge    340000.0         NaN  
...                                     ...         ...         ...  
1050  DSG Consumer Partners, Blume Ventures   1500000.0         NaN  
1051          Sequoia India, Ribbit Capital  13200000.0  Seed Round  
1052                             Verlinvest   8000000.0         NaN  
1053             SIG Global India Fund LLP.   8043000.0    Series C  
1054                      Chiratae Ventures   9000000.0    Series A  

[914 rows x 7 columns]

In [70]:
#Checking the missing values in the new dataframe d20_cleaned1
d20_cleaned1.isna().sum()

Company_Brand      0
HeadQuarter       85
Sector            12
Founders          11
Investor          31
Amount           113
Stage            323
dtype: int64

In [75]:
# Impute missing values
d20_cleaned1['HeadQuarter'].fillna(d20_cleaned1['HeadQuarter'].mode()[0], inplace=True)
d20_cleaned1['Sector'].fillna(d20_cleaned1['Sector'].mode()[0], inplace=True)
d20_cleaned1['Founders'].fillna('Unknown', inplace=True)
d20_cleaned1['Investor'].fillna('Unknown', inplace=True)
d20_cleaned1['Amount'].fillna(d20_cleaned1['Amount'].median(), inplace=True)
d20_cleaned1['Stage'].fillna('Unknown', inplace=True)

Company_Brand               HeadQuarter           Sector  \
21       SucSEED Indovation                 Hyderabad          FinTech   
24         Circle of Angels                   Gurgaon          FinTech   
151               BlackSoil                    Mumbai          FinTech   
241                SoOLEGAL                 New Delhi        LegalTech   
262          Phononic Vibes  Milano, Lombardia, Italy         Deeptech   
274                 Antaios                    France     Tech company   
305         Sapio Analytics                    Mumbai     Tech company   
318               GenieTalk                    Indore     Tech company   
325                 BestDoc                     Kochi       HealthCare   
329       Inflexor Ventures                    Mumbai          FinTech   
335                 SignalX                 Hyderabad    SaaS platform   
337  Vesta Space Technology                      Pune     Tech Startup   
338                   Dozee                 Bangalore       HealthTech   
346           3one4 Capital                 Bangalore  Venture capital   
365                  QShala                 Bangalore           EdTech   

                                              Founders Investor       Amount  \
21                                    Vikrant Varshney      NaN    5000000.0   
24                                      Karanpal Singh      NaN    3000000.0   
151                                       Ankur Bansal      NaN   10000000.0   
241                                        Manish Kaul      NaN    4000000.0   
262  Andrea Boeri, Braghin Francesco, Giovanni Cape...      NaN    2300000.0   
274                               Jean Pierre Nozieres      NaN   11000000.0   
305                   Ashwin Srivastava, Hardik Somani      NaN     150000.0   
318          Santosh Kataria, Ankit Kimtee, Vivek Jain      NaN     750000.0   
325                Afsal Salu, Fayaz Bin Abdu, Soudabi      NaN    2100000.0   
329  Parampara Capital, Venkat Vallabhaneni, Jatin ...      NaN   10000000.0   
335                                Govind Balachandran      NaN     750000.0   
337                                 Arun Kumar Sureban      NaN   10000000.0   
338                                    Gaurav Parchani      NaN    1600000.0   
346                           Pranav Pai, Siddarth Pai      NaN  100000000.0   
365                   Sachin Ravi, Raghav Chakravarthy      NaN     300000.0   

    Stage  
21    NaN  
24    NaN  
151   NaN  
241   NaN  
262   NaN  
274   NaN  
305   NaN  
318   NaN  
325   NaN  
329   NaN  
335   NaN  
337   NaN  
338   NaN  
346   NaN  
365   NaN

In [76]:
# Impute missing values
d20_cleaned1['HeadQuarter'].fillna(d20_cleaned1['HeadQuarter'].mode()[0], inplace=True)
d20_cleaned1['Sector'].fillna(d20_cleaned1['Sector'].mode()[0], inplace=True)
d20_cleaned1['Founders'].fillna('Unknown', inplace=True)
d20_cleaned1['Investor'].fillna('Unknown', inplace=True)
d20_cleaned1['Amount'].fillna(d20_cleaned1['Amount'].median(), inplace=True)
d20_cleaned1['Stage'].fillna('Unknown', inplace=True)

In [77]:
#CVerifying that there are no missing values after the imputation process
d20_cleaned1.isna().sum()

Company_Brand    0
HeadQuarter      0
Sector           0
Founders         0
Investor         0
Amount           0
Stage            0
dtype: int64

### Restructuring Data
After taking care of the missing values, there is still work to be done to make the dataset presentable and usable. This include renaming of columns to ensure that when merging with the other dataset, we havve identical column headers. Also, a column named Year will be added to be able to distinguish between rows from different dataset year when the merging occurs. 

In [80]:
#Renaming the columns in the DataFrame d20_cleaned1
d20_cleaned1.rename(columns={'Company_Brand' : 'Company_Name'}, inplace=True)
d20_cleaned1.rename(columns={'HeadQuarter' : 'Location'}, inplace=True)
d20_cleaned1.rename(columns={'Sector' : 'Industry'}, inplace=True)
d20_cleaned1.rename(columns={'Stage' : 'Funding_Stage'}, inplace=True)


In [81]:
#Adding a colomn named 'Year' to the dataframe d20_cleaned1
d20_cleaned1['Year']= 2020
d20_cleaned1

Company_Name   Location            Industry  \
0       Aqgromalin    Chennai            AgriTech   
1         Krayonnz  Bangalore              EdTech   
2     PadCare Labs       Pune  Hygiene management   
3            NCOME  New Delhi              Escrow   
4       Gramophone     Indore            AgriTech   
...            ...        ...                 ...   
1050  Leverage Edu      Delhi              Edtech   
1051         EpiFi  Bangalore             Fintech   
1052       Purplle     Mumbai           Cosmetics   
1053        Shuttl      Delhi           Transport   
1054         Pando    Chennai            Logitech   

                                               Founders  \
0                       Prasanna Manogaran, Bharani C L   
1                      Saurabh Dixit, Gurudutt Upadhyay   
2                                       Ajinkya Dhariya   
3                                         Ritesh Tiwari   
4     Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
...                                                 ...   
1050                                  Akshay Chaturvedi   
1051                    Sujith Narayanan, Sumit Gwalani   
1052                          Manish Taneja, Rahul Dash   
1053                      Amit Singh, Deepanshu Malviya   
1054                     Jayakrishnan, Abhijeet Manohar   

                                   Investor      Amount Funding_Stage  Year  
0                           Angel investors    200000.0       Unknown  2020  
1                           GSF Accelerator    100000.0      Pre-seed  2020  
2                            Venture Center   3000000.0      Pre-seed  2020  
3       Venture Catalysts, PointOne Capital    400000.0       Unknown  2020  
4       Siana Capital Management, Info Edge    340000.0       Unknown  2020  
...                                     ...         ...           ...   ...  
1050  DSG Consumer Partners, Blume Ventures   1500000.0       Unknown  2020  
1051          Sequoia India, Ribbit Capital  13200000.0    Seed Round  2020  
1052                             Verlinvest   8000000.0       Unknown  2020  
1053             SIG Global India Fund LLP.   8043000.0      Series C  2020  
1054                      Chiratae Ventures   9000000.0      Series A  2020  

[914 rows x 8 columns]

In [82]:
#Viewing a summary on d20_cleaned1 to ensure that the columns have the desired datatypes
d20_cleaned1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 914 entries, 0 to 1054
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Name   914 non-null    object 
 1   Location       914 non-null    object 
 2   Industry       914 non-null    object 
 3   Founders       914 non-null    object 
 4   Investor       914 non-null    object 
 5   Amount         914 non-null    float64
 6   Funding_Stage  914 non-null    object 
 7   Year           914 non-null    int64  
dtypes: float64(1), int64(1), object(6)
memory usage: 64.3+ KB


### Handling Outliers
The presence of outliers can skew th eresults of our statisical analyses and as such we need to identify and handle/manage them. In this section, we study the presence of outliers in the Amount column of our dataset.


In [85]:
#Calculate the First and Third Quartiles (Q1 and Q3):
Q1 = d20_cleaned1['Amount'].quantile(0.25)
Q3 = d20_cleaned1['Amount'].quantile(0.75)

#Calculate the Interquartile Range (IQR)
IQR = Q3 - Q1

# Identify outliers
outliers = d20_cleaned1[((d20_cleaned1['Amount'] < (Q1 - 1.5 * IQR)) | (d20_cleaned1['Amount'] > (Q3 + 1.5 * IQR)))]
outliers


Company_Name   Location                Industry  \
8               Rupeek  Bangalore                 FinTech   
15              Zomato    Gurgaon           Food Delivery   
26              Glance  Bangalore              AI startup   
27    VerSe Innovation  Bangalore       Telecommunication   
45               Ayana  Bangalore        Renewable player   
...                ...        ...                     ...   
1009            Bounce  Bangalore  Automotive and Rentals   
1010            Zinier  Bangalore              Automation   
1034            Zomato  Bangalore                Foodtech   
1035              Byju  Bangalore                  Edtech   
1037        HighRadius  Hyderabad                    SaaS   

                                             Founders  \
8             Amar Prabhu, Ashwin Soni, Sumit Maniyar   
15      Deepinder Goyal, Gaurav Gupta, Pankaj Chaddah   
26                                      Naveen Tewari   
27                                     Virendra Gupta   
45                                        Shivanand N   
...                                               ...   
1009  Vivekananda Hallekere, Anil Giri Raju,Arun Agni   
1010                                        Arka Dhar   
1034                                  Deepinder Goyal   
1035                                  Byju Raveendran   
1037                                   Sashi Narahari   

                                          Investor       Amount Funding_Stage  \
8     KB Investment, Bertelsmann India Investments   45000000.0      Series C   
15                   Kora, Tiger Global Management  660000000.0       Unknown   
26                     Mithril Capital Management.  145000000.0       Unknown   
27                                          Google  100000000.0       Unknown   
45                             NIIF Ltd, CDC Group  390000000.0       Unknown   
...                                            ...          ...           ...   
1009                     Accel Partners, B Capital   97500000.0      Series D   
1010                                ICONIQ Capital   90000000.0      Series C   
1034                                 Ant Financial  150000000.0       Unknown   
1035                                  Tiger Global  200000000.0       Unknown   
1037                                ICONIQ Capital  125000000.0      Series B   

      Year  
8     2020  
15    2020  
26    2020  
27    2020  
45    2020  
...    ...  
1009  2020  
1010  2020  
1034  2020  
1035  2020  
1037  2020  

[131 rows x 8 columns]

In [89]:
ods=  outliers[outliers["Funding_Stage"] == 'Unknown']
ods.info()

<class 'pandas.core.frame.DataFrame'>
Index: 58 entries, 15 to 1035
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Name   58 non-null     object 
 1   Location       58 non-null     object 
 2   Industry       58 non-null     object 
 3   Founders       58 non-null     object 
 4   Investor       58 non-null     object 
 5   Amount         58 non-null     float64
 6   Funding_Stage  58 non-null     object 
 7   Year           58 non-null     int64  
dtypes: float64(1), int64(1), object(6)
memory usage: 4.1+ KB


#### Conclusion on Outliers
There are 131 rows in the Amount column that were outliers. Out of the 131, 58 of them had an Unknwon funding stage. Initially, the thought was to remove that subset (outliers that have missing/unknwown Funding Stage) but that would affect other columns such as Locaton as a majority (25) of this subset were located at Bangalore and thus removing these outliers could affect the data.

### String Data Cleaning

For some of the entries in our data, they may be in the wrong format or would have wrong characters added to them that can cause issues when we are handling and manipulating the data thus it is necessary to correct any noise (messy/incorrect string text). This will ensure consistencyy an accuracy as we perform our analysis.

In [96]:
#Listing all the unique values in the Location column
d20_cleaned1['Location'].unique()

array(['Chennai', 'Bangalore', 'Pune', 'New Delhi', 'Indore', 'Hyderabad',
       'Gurgaon', 'Belgaum', 'Mumbai', 'Noida', 'Andheri', 'Ahmedabad',
       'Jaipur', 'Kolkata', 'Tirunelveli, Tamilnadu', 'Thane',
       'Singapore', 'Gurugram', 'Gujarat', 'Haryana', 'Jodhpur',
       'Jaipur, Rajastan', 'Delhi', 'California', 'Dhingsara, Haryana',
       'New York, United States',
       'San Francisco, California, United States',
       'San Francisco, United States', 'San Ramon, California',
       'Paris, Ile-de-France, France', 'Plano, Texas, United States',
       'Sydney', 'San Francisco Bay Area, Silicon Valley, West Coast',
       'Bangaldesh', 'London, England, United Kingdom',
       'Sydney, New South Wales, Australia', 'Milano, Lombardia, Italy',
       'France', 'San Francisco Bay Area, West Coast, Western US',
       'Trivandrum, Kerala, India', 'Cochin', 'Samastipur, Bihar',
       'Irvine, California, United States', 'Tumkur, Karnataka',
       'Newcastle Upon Tyne, Newcas

In [108]:
#Creating a variable for the list of incorrect locations
wrong_location = ['Singapore', 'California',
'San Ramon, California', 'Paris, Ile-de-France, France', 'Plano, Texas, United States',
'Sydney', 'San Francisco Bay Area, Silicon Valley, West Coast', 'Bangladesh',
'London, England, United Kingdom', 'Sydney, New South Wales, Australia',
'Milano, Lombardia, Italy', 'France', 'San Francisco Bay Area, West Coast, Western US',
'Irvine, California, United States', 'Newcastle Upon Tyne, Newcastle upon Tyne, United Kingdom',
'Shanghai, China', 'Jiaxing, Zhejiang, China', 'San Francisco',
'Tangerang, Jawa Barat, Indonesia', 'Berlin', 'Seattle', 'Riyadh', 'Seoul', 'New York',
'Bangkok','New York, United States',  'San Francisco, California, United States', 'San Francisco, United States', 'Bangaldesh']

#display rows in d20cleaned1 that have startups not from India
d20_cleaned1[d20_cleaned1['Location'].isin(wrong_location)]

Company_Name                                           Location  \
91                  Kyt                                          Singapore   
176             Udacity                                         California   
186            Leena AI                            New York, United States   
223              Hubilo           San Francisco, California, United States   
225               Ketos                       San Francisco, United States   
229              Tekion                              San Ramon, California   
230  Brighteye Ventures                       Paris, Ile-de-France, France   
232           VitalTech                        Plano, Texas, United States   
239             Tiliter                                             Sydney   
240             Slintel  San Francisco Bay Area, Silicon Valley, West C...   
245              ShopUp                                         Bangaldesh   
260             Arrival                    London, England, United Kingdom   
261        Cover Genius                 Sydney, New South Wales, Australia   
262      Phononic Vibes                           Milano, Lombardia, Italy   
264               Clair                            New York, United States   
273             Your MD                    London, England, United Kingdom   
274             Antaios                                             France   
279           Chargebee     San Francisco Bay Area, West Coast, Western US   
287      Monet Networks                  Irvine, California, United States   
292              Omnify  San Francisco Bay Area, Silicon Valley, West C...   
300           Coralogix           San Francisco, California, United States   
301            Equiwatt  Newcastle Upon Tyne, Newcastle upon Tyne, Unit...   
303            Zhangmen                                    Shanghai, China   
306              Taimei                           Jiaxing, Zhejiang, China   
317           Snowflake                                         California   
373          SquadStack                                      San Francisco   
384          Yummy Corp                   Tangerang, Jawa Barat, Indonesia   
386          Clairvolex                                         California   
388             Brinton                                         California   
390              Infarm                                             Berlin   
391               Zuper                                            Seattle   
397               Penny                                             Riyadh   
399             Precily                                         California   
400          SmartClean                                          Singapore   
413            DillyHub                                              Seoul   
420          Observe ai                                      San Francisco   
456         Hippo Video                                           New York   
457              Hasura                                      San Francisco   
458              Qapita                                          Singapore   
459              Sunday                                            Bangkok   
483            CredoLab                                          Singapore   

                  Industry                                           Founders  \
91                  EdTech                        Bhavik Rathod, Tripti Ahuja   
176                 EdTech      David Stavens, Mike Sokolsky, Sebastian Thrun   
186             AI startup           Adit Jain, Anand Prajapati, Mayank Goyal   
223           Tech company                       Mayank Agarwal, Vaibhav Jain   
225             AI startup                                     Meena Sankaran   
229             AI startup                     Guru Sankararaman, Jay Vijayan   
230                 Edtech                                       David guerin   
232              HeathTech                               Anjan Panneer Selvam   
239  

In [109]:
# Creating a mask to isolate the rows with correct locations
mask = ~d20_cleaned1['Location'].isin(wrong_location)

# Removing rows where startups are not located in India and creating a new DataFrame d20_cleaned2
d20_cleaned2 = d20_cleaned1[mask]


d20_cleaned2.info()

<class 'pandas.core.frame.DataFrame'>
Index: 873 entries, 0 to 1054
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Name   873 non-null    object 
 1   Location       873 non-null    object 
 2   Industry       873 non-null    object 
 3   Founders       873 non-null    object 
 4   Investor       873 non-null    object 
 5   Amount         873 non-null    float64
 6   Funding_Stage  873 non-null    object 
 7   Year           873 non-null    int64  
dtypes: float64(1), int64(1), object(6)
memory usage: 61.4+ KB


In [110]:
#Listing the unique locations after removing startups not in India
d20_cleaned2['Location'].unique()

array(['Chennai', 'Bangalore', 'Pune', 'New Delhi', 'Indore', 'Hyderabad',
       'Gurgaon', 'Belgaum', 'Mumbai', 'Noida', 'Andheri', 'Ahmedabad',
       'Jaipur', 'Kolkata', 'Tirunelveli, Tamilnadu', 'Thane', 'Gurugram',
       'Gujarat', 'Haryana', 'Jodhpur', 'Jaipur, Rajastan', 'Delhi',
       'Dhingsara, Haryana', 'Trivandrum, Kerala, India', 'Cochin',
       'Samastipur, Bihar', 'Tumkur, Karnataka', 'Rajastan', 'Kochi',
       'Ludhiana', 'Dehradun', 'Kanpur', 'Chandigarh', 'Hyderebad',
       'Odisha', 'Bihar', 'Goa', 'Uttar Pradesh', 'Banglore',
       'Coimbatore'], dtype=object)

In [112]:
#replacing locations with multiple names with a common name for consistency
replacements = {
    'Banglore': 'Bangalore',
    'Gurugram': 'Gurgaon',
    'Jaipur, Rajastan': 'Jaipur',
    'Cochin': 'Kochi'
}
d20_cleaned2.replace(replacements, inplace=True)


#Listing the unique locations to verify the replacement were successful
d20_cleaned2['Location'].unique()

array(['Chennai', 'Bangalore', 'Pune', 'New Delhi', 'Indore', 'Hyderabad',
       'Gurgaon', 'Belgaum', 'Mumbai', 'Noida', 'Andheri', 'Ahmedabad',
       'Jaipur', 'Kolkata', 'Tirunelveli, Tamilnadu', 'Thane', 'Gujarat',
       'Haryana', 'Jodhpur', 'Delhi', 'Dhingsara, Haryana',
       'Trivandrum, Kerala, India', 'Kochi', 'Samastipur, Bihar',
       'Tumkur, Karnataka', 'Rajastan', 'Ludhiana', 'Dehradun', 'Kanpur',
       'Chandigarh', 'Hyderebad', 'Odisha', 'Bihar', 'Goa',
       'Uttar Pradesh', 'Coimbatore'], dtype=object)